src/rust/engine/process_execution/src/remote.rs

```diff
@@ -844,15 +846,80 @@ fn maybe_add_workunit(
   }
 }
 
-pub fn make_execute_request(
+fn make_wrapper_for_append_only_caches(
+  caches: &BTreeMap<CacheName, RelativePath>,
+  base_path: &str,
+) -> Result<String, String> {
+  let mut script = String::new();
+  writeln!(&mut script, "#!/bin/sh").map_err(|err| format!("write! failed: {err:?}"))?;
+  for (cache_name, path) in caches {
+    writeln!(
+      &mut script,
+      "mkdir -p '{}/{}'",
```

Comment:
Do you know how this will interact with Pants messing with the PATH sometimes?

Reply:
This should probably get the `mkdir` and `/bin/sh` paths from a `BinaryPathRequest`.

Reply:
Or at least hardcode to `/bin/mkdir` for an initial implementation. Or maybe let users define the paths in `remote_environment` targets (to accommodate Docker containers)? Using `BinaryPathRequest` may be hard to do with chicken and egg?

Reply:
Yes, definitely a "chicken and egg" problem. it is unclear to me how to trigger a `BinaryPathRequest` from this part of the Rust code.

Reply:
Another open question: Should failure to symlink the append-only cache fail the `Process` as well?

Reply:
Maybe a failure fails all of Pants, and we instruct the user to disable this mechanism by unsettling the global option? We could mention that they can open a GitHub issue if they need more flexibility like setting `/bin/mkdir`